pkg/resolve/selector.go

```diff
@@ -0,0 +1,116 @@
+package resolve
+
+import (
+	"bytes"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/labels"
+	"k8s.io/apimachinery/pkg/runtime"
+	yaml2json "k8s.io/apimachinery/pkg/util/yaml"
+	"regexp"
+	"sigs.k8s.io/yaml"
+	"strings"
+)
+
+
+// FilterBySelector filters out any resources
+// from the raw manifest bytes whose labels
+// don't match the provided selector
+func FilterBySelector(input []byte, selectorString string) ([]byte, error) {
+	selector, err := labels.Parse(selectorString)
+	if err != nil {
+		return nil, err
+	}
+
+	var outputObjectsYaml [][]byte
+
+	// helper function to append selected objects to our result
+	appendObj := func(obj runtime.Object) error {
+		rawJson, err := runtime.Encode(unstructured.UnstructuredJSONScheme, obj)
+		if err != nil {
+			return err
+		}
+		rawYaml, err := yaml.JSONToYAML(rawJson)
+		if err != nil {
+			return err
+		}
+		outputObjectsYaml = append(outputObjectsYaml, rawYaml)
+		return nil
+	}
+
+	// parse runtime.Objects from the input yaml
+	objects, err := parseUnstructured(input)
+	if err != nil {
+		return nil, err
+	}
+
+	for _, object := range objects {
+		// objects parsed by UnstructuredJSONScheme can only be of
```

Comment:
We might want to log something or panic in a default case in case this changes, just to make sure we're not silently dropping documents?